feat(kfileupload): component reskin [KHCP-8982] (#1926)
Kong/kongponents: docs/components/file-upload.md

# File Upload

KFileUpload provides a wrapper around an `input` element with `type=file` for file upload.

<KFileUpload label="File upload" :label-attributes="{ info: `Accepted file types: ${acceptedFileType.join(', ')}` }" help="File size must be less than 1MB." :accept="acceptedFileType" />

```html
<KFileUpload
  label="Upload File"
  :label-attributes="{ info: 'Accepted file types: .yml, .yaml, .json, .md, .markdown, image/*' }"
  :accept="acceptedFileType"
/>
```

## Props

### accept

> Please refer to [mdn web docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#accept) that defines the file types the file input should accept.

Reviewer: ```suggestion
Please refer to [mdn web docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#accept) that defines the file types the component can accept.
```

### errorMessage

String to be displayed as error message if `hasError` prop is `true`.

<KFileUpload label="Upload file" error error-message="File size must be less than 1MB." :accept="acceptedFileType" />

```html
<KFileUpload
  label="Upload file"
  error
  error-message="File size must be less than 1MB."
  :accept="acceptedFileType"
/>
```

### placeholder

Use the `placeholder` prop to display placeholder text.

<KFileUpload placeholder="Select file on your computer" :accept="acceptedFileType" />

```html
<KFileUpload placeholder="Select file on your computer" :accept="acceptedFileType" />
```

### buttonText

This is the text that will be displayed on the button that triggers the click on KInput.

<KFileUpload button-text="Choose file" :accept="acceptedFileType" />

```html
<KFileUpload button-text="Choose file" :accept="acceptedFileType" />
```

### maxFileSize

Use this prop to customize the maximize size of file that can be uploaded. Default value for `image` upload is `1MB` and for `file` is `5.25MB`.

<KFileUpload :max-file-size="2" :accept="['.jpg', '.png']" />

```html
<KFileUpload :max-file-size="2" :accept="acceptedFileType" />
```

:::tip NOTE

> By default KFileUpload will display error state with a generic error message when selected file size exceeds specified maximum file size. You can use [`errorMessage` prop](#errormessage) in conjunction with [`error` event](#events) to display custom error message.

Reviewer: ```suggestion
By default KFileUpload will display the error state with a generic error message when selected file exceeds specified maximum file size. You can use [`errorMessage` prop](#errormessage) in conjunction with the [`error` event](#events) to display a custom error message.
```

- `type` (returns the MIME type of the file)
- `webkitRelativePath` (returns the path the URL of the File is relative to)

<KFileUpload label="Upload file" :label-attributes="{ info: `Accepted file types: ${acceptedFileType.join(', ')}` }" :accept="acceptedFileType" @file-added="file => printData(file)" @file-removed="() => fileData = ''" />

```html
<KFileUpload
  label="Upload file"
  @file-added="file => printData(file)"
  @file-removed="() => fileData = ''"
/>
```

<br/>
<div>Emitted value:

>   <pre v-if="fileData.length" class="emitted-value">

Reviewer: suggestion: can you add a background color for this `pre` element just on this docs page so we can tell it's displaying something?
